G2P.pm

```diff
@@ -735,26 +735,22 @@ sub gene_overlap_filtering {
   if (! defined $pass_gene_overlap_filter && ! defined $self->{user_params}->{filter_by_gene_symbol}) {
     my @gene_hgnc = split /:/, $transcript->{_gene_hgnc_id} if (defined $transcript->{_gene_hgnc_id});
     my $hgnc_id = $gene_hgnc[1] if (@gene_hgnc);
-    foreach my $gene_id ($hgnc_id){
-      foreach my $id (keys $self->{hgnc_mapping}) {
-        if ( defined($gene_id) && $gene_id == $id) {
-          my $gene_symbol = $self->{hgnc_mapping}{$id};
-          my $gene_data = $self->gene_data($gene_symbol) if defined ($gene_symbol);
-          if (defined $gene_data) {
-            if (defined $gene_data->{'allelic requirement'} && scalar @{$gene_data->{'allelic requirement'}}) {
-              foreach my $ar (@{$gene_data->{'allelic requirement'}}) {
-                $self->{ar}->{$gene_stable_id}->{$ar} = 1;
-              }
-              # this is for the log data, G2P gene data is called in the write report 
-              # added in the write_report method also 
-              $self->write_report('G2P_gene_data', $gene_stable_id, $gene_data, $gene_data->{'gene_xrefs'}, $gene_data->{'HGNC'}, $gene_data->{'confidence_category'}, $gene_data->{'confidence_value'} );
+    foreach my $id (keys $self->{hgnc_mapping}) {
+      if ( defined($hgnc_id) == $id) {
```

Comment:
@olaaustine - are you sure this works? Isn't $id compared to 0 or 1 depending on whether $hgnc_id is defined or not?